### PR TITLE
Return `CompletableFuture` instead of `manifold/deferred`

### PR DESCRIPTION
**Pull request by barkanido**
Use the more lightweight, `promesa` lib instead of manifold. All API's now return a `CompletableFuture` and users can then use either Clojure (`future`, `deref` etc) or java interop directly or manifold/deferred or `promesa` in order to later compose more async behavior.

### CODE REVIEW

File: src/main/clojure/aerospike_clj/client.clj [hunk 1]
```diff
@@ -96,77 +96,80 @@
...
-  op-future)
+  (let [reducer 
+        (fn
+          ([] op-future)
+          ([op-future ce]
+           (-> op-future
+               (p/then (fn [op-result]
+                         (on-success  ce op-name op-result index op-start-time db))) 
+               (p/catch (fn [op-result]
+                          (on-success ce op-name op-result index op-start-time db))))))]
```
Comment (haski) [inline src/main/clojure/aerospike_clj/client.clj]: this one should be onFailure

Comment (barkanido) [reply src/main/clojure/aerospike_clj/client.clj]: right!

File: src/main/clojure/aerospike_clj/client.clj [hunk 2]
```diff
@@ -102,9 +102,9 @@
           ([op-future ce]
            (-> op-future
                (p/then (fn [op-result]
-                         (on-success  ce op-name op-result index op-start-time db))) 
-               (p/catch (fn [op-result]
-                          (on-success ce op-name op-result index op-start-time db))))))]
+                         (on-success  ce op-name op-result    index op-start-time db))) 
+               (p/catch (fn [op-exception]
```
Comment (haski) [inline src/main/clojure/aerospike_clj/client.clj:106]: you're still calling on-success from p/catch where you should be calling on-failure

Comment (barkanido) [reply src/main/clojure/aerospike_clj/client.clj:106]: oops. fixed, once again.

**Review by haski — APPROVED** [review]

**Pull request merged**